keps/sig-node/4328-node-label-manager/README.md

```diff
@@ -0,0 +1,404 @@
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial unit/e2e tests completed and enabled
+- Documentation is added to demonstrate why this is useful how exactly affinity needs to be configured.
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Additional e2e tests are completed(if needed)
+
+#### GA
+
+- No negative feedback
+- No bug issues reported
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+The previous node affinity will not be broken. Users can continue to use their affinity as it is.
+To use this enhancement, users need to enable the feature gate, and add `RequiredDuringSchedulingRequiredDuringExecution` field for pod's node affinity.
+
+#### Downgrade
+The `RequiredDuringSchedulingRequiredDuringExecution` field will be ignored during both the scheduling phase and the execution phase. 
+
+### Version Skew Strategy
+
+kube-controller-manager, kube-scheduler and kube-apiserver will need enable the feature gate for the full featureset
+to be present. 
+
+If kube-apiserver doesn't enable the feature gate, this new affinity will be ignored.
+
+If kube-apiserver enable the feature gate, and:
+if only the kube-controller-manager enables the feature gate, the node affinity will be ignored during the scheduling phase.
+if only the kube-scheduler enables the feature gate, the node affinity will be ignored during the execution phase. 
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NodeAffinityController 
+  - Components depending on the feature gate: kube-controller-manager, kube-scheduler, kube-apiserver
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+The feature can be disabled in Alpha and Beta versions by restarting kube-controller-manager, kube-scheduler and kube-apiserver with the feature-gate off. 
+In terms of Stable versions, users can choose to opt-out by not setting the corresponding field.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The `RequiredDuringSchedulingRequiredDuringExecution` node affinity will take effect as expected.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The change is opt-in, it doesn't impact already running workloads. 
```

Comment:
This questions asks about disabling the feature after it was enabled and workloads are using the feature.